src/logic/wallets/store/actions/fetchProvider.js

```diff
@@ -44,10 +44,24 @@ const handleProviderNotification = (openSnackbar: Function, provider: ProviderPr
   openSnackbar(msg, variant)
 }
 
+export const recurrentFetchProvider = (currentProvider: ProviderProps, openSnackbar: Function) => async (
+  dispatch: ReduxDispatch<*>,
+) => {
+  const newProvider: ProviderProps = await getProviderInfo()
+
+  if (JSON.stringify(currentProvider) !== JSON.stringify(newProvider)) {
```

Comment:
Since its synchronous and will be run every 2s probably we can write a `simpleStringify` function? I found this benchmark
https://www.measurethat.net/benchmarks/ShowResult/9266 where `simpleStringify` is almost 3x faster for me.